blog/2025-05-02-mobile-override-search-area.mdx

```diff
@@ -0,0 +1,172 @@
+---
+title: Override Search Area in VertiGIS Studio Mobile
+author: Jack Basha
+authorTitle: Quality Assurance Analyst
+authorURL: https://github.com/jackbasha
+authorImageURL: https://github.com/jackbasha.png
+tags: [mobile, search]
+---
+
+import Link from "@docusaurus/Link";
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+VertiGIS Studio Mobile enables app developers to override the default search area, which is the current visible area of the map when the app is not backed by an MMPK, to a custom area (extent) of the developer's choosing. This requires hand-editing the app JSON and working out a few details. In this blog post, I will guide you on how to do that and how to define a custom extent for search to operate within.
+
+<!--truncate-->
+
+## 1 - Finding the Area to Override Search
+
+To start off the work, we'll need to head to VertiGIS Studio Workflow and create a new blank Workflow. This Workflow will be used to override the search functionality in Mobile.
+
+There are two ways of obtaining the extent numbers needed to override the search area: Using a layer's FullExtent property to obtain the extent, or using a feature action on a polygon feature to create a custom search area.
+
+I will walk you through how to use both of these methods to enable the most amount of customization.
+
+### 1.1 - Using a layer's FullExtent property to obtain the extent {#using-layer-bounds}
+
+Using a layer to obtain the extent numbers is the more foolproof way since it doesn't involve hard-coding any specific numbers to Workflow.
+
+:::info Note
+
+This is the recommended way as using a feature to obtain the extent numbers is a more complicated process.
+
+:::
+
+Use the `Get Layer` activity to obtain a layer that will act as the basis of the search area. While testing, I found this to cover a much larger area than what's actually covered by the features of the layer so, although this works to extend the search area, developers who want a finer control of the search area override should use the method described in [section 1.2](#using-feature-bounds).
+
+Assuming the `Get Layer` activity's ID is `layer1`, we will be using `=$layer1.layer.Unwrap().FullExtent` as input to the `Set Property` activity. This is an undocumented function because it unwraps a layer from what we use in VertiGIS Studio products into ESRI's layer objects, so unwrapping layers should be done with care.
```

Comment:
```suggestion
Assuming the `Get Layer` activity's ID is `layer1`, we will be using `=$layer1.layer.Unwrap().FullExtent` as input to the `Set Property` activity. This is an undocumented function which unwraps a layer from what we use internally in VertiGIS Studio products into ESRI's layer objects.
```
The "should be done with care" bit sounds slightly scarier than it needs to be, since it doesn't exactly have any negative side effects.